Sources/Datadog/URLSessionAutoInstrumentation/DDURLSessionDelegate.swift

```diff
@@ -6,12 +6,28 @@
 
 import Foundation
 
+/// In case that you are not able to use `DDURLSessionDelegate` or its subclasses in your `URLSession` instance,
+/// you can use `DDURLSessionDelegateProviding` protocol.
+/// Implement this protocol in your `URLSessionDelegate` class and forward required API calls to `DDURLSessionDelegate`.
+@objc
+public protocol URLSessionDelegateProviding: URLSessionDelegate {
```

Comment:
We could additionally prefix this name with `__` (`__URLSessionDelegateProviding`), which would emphasize its internal use. And the protocol comment could be less-encouraging for the use - IMO sth less direct may work better:
```swift
/// An interface for forwarding `URLSessionDelegate` calls to `DDURLSessionDelegate`.
/// The implementation must ensure that required methods are called on the `ddURLSessionDelegate`.
```